File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> arr) {
    int answer = 0;

    while (true) {
        vector<int> copy = arr;

        for (int& num : arr) {
            if (num >= 50 && num % 2 == 0) {
                num /= 2;
            }
            if (num < 50 && num % 2 != 0) {
                num *= 2;
                num += 1;
            }
        }

        if(copy == arr) {
            break;
        }
        
        answer++;
    }

    return answer;
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.swift
import Foundation

func solution(_ arr:[Int]) -> Int {
    var afterArr:[Int] = []
    var beforeArr = arr
    var result = 0
    
    while true {
        afterArr = beforeArr.map { num  in
            if num.isMultiple(of: 2) && num >= 50 { return num / 2 }
            else if !num.isMultiple(of: 2) && num < 50 { return num * 2 + 1 }
            else { return num }
        }
        if afterArr.elementsEqual(beforeArr) { break }
        beforeArr = afterArr
        result += 1
    }
    
    return result
}